pymc_experimental/statespace/core/statespace.py

```diff
@@ -0,0 +1,1296 @@
+                    f"{filter_output}_{group}",
+                    mus=mus,
+                    covs=covs,
+                    logp=pt.zeros(mus.shape[0]),
+                    dims=mu_dims,
+                )
+            idata_conditional = pm.sample_posterior_predictive(
+                group_idata,
+                var_names=[f"filtered_{group}", f"predicted_{group}", f"smoothed_{group}"],
+                compile_kwargs={"mode": get_mode(self._fit_mode)},
+            )
+        return idata_conditional.posterior_predictive
+
+    def _sample_unconditional(
+        self,
+        idata: InferenceData,
+        group: str,
+        steps: Optional[int] = None,
+        use_data_time_dim: bool = False,
+    ):
+        """
+        Draw unconditional sample trajectories according to state space dynamics, using random samples from the
+        a provided trace. The state space update equations are:
+
+            X[t+1] = T @ X[t] + R @ eta[t], eta ~ N(0, Q)
+            Y[t] = Z @ X[t] + nu[t], nu ~ N(0, H)
+            x[0] ~ N(a0, P0)
+
+        Parameters
+        ----------
+        idata : InferenceData
+            An Arviz InferenceData object with a posterior group containing samples from the
+            posterior distribution over model parameters.
+
+        steps : Optional[int], default=None
+            The number of time steps to sample for the unconditional trajectories. If not provided (None),
+            the function will sample trajectories for the entire available time dimension in the posterior.
+            Otherwise, it will generate trajectories for the specified number of steps.
+
+        use_data_time_dim : bool, default=False
+            If True, the function uses the time dimension present in the provided `idata` object to sample
+            unconditional trajectories. If False, a custom time dimension is created based on the number of steps
+            specified, or if steps is None, it uses the entire available time dimension in the posterior.
+
+        Returns
+        -------
+        InferenceData
+            An Arviz InfereceData with two groups, posterior_latent and posterior_observed
+
+            - posterior_latent represents the latent state trajectories `X[t]`, which follows the dynamics:
+              `x[t+1] = T @ x[t] + R @ eta[t]`, where `eta ~ N(0, Q)`.
+
+            - posterior_observed represents the observed state trajectories `Y[t]`, which is obtained from
+              the latent state trajectories: `y[t] = Z @ x[t] + nu[t]`, where `nu ~ N(0, H)`.
+        """
+        _verify_group(group)
+        group_idata = getattr(idata, group)
+        dims = None
+        temp_coords = self._fit_coords.copy()
+
+        if not use_data_time_dim and steps is not None:
+            temp_coords.update({TIME_DIM: np.arange(1 + steps, dtype="int")})
+            steps = len(temp_coords[TIME_DIM]) - 1
+        elif steps is not None:
+            n_dimsteps = len(temp_coords[TIME_DIM])
```

Comment:
Yeah this is a rehash of that shape vs step conversation we had when I was doing the MarkovChain distribution, which admittedly I totally forgot about. I guess i really have my heart set on the idea that if I ask for 10 steps, I should get back a distribution with shape (10, k). I could change this argument to be called "shape" instead of "steps" to match the `distributions.timeseries` api?

In the statespace framework, the more lags makes the state vector larger, but there's always just one initial state, so no need to account for a variable number of initial states.

Reply:
steps is mostly internal like in timeseries no? 

Most users will use shape/dims so I think it's better to stay consistent with timeseries meaning just for developers sake?

I am not a big fan of the step/shape distinction either btw

Reply:
Yeah steps is internal everywhere in the API.

This shows up here because when you do `sample_unconditional`, you can generate data for as long as you like. By default it matches the data's length, but the data isn't used at all in the computation, so I need to put something there for a "simulation length" argument. I guess it could be "simulation length".